server/jobs.coffee

```diff
@@ -32,10 +32,7 @@ class @WarrantyLookupJob extends Job
           accept: 'application/json'
           apikey: Meteor.settings.apiKeys.dellWarranty
       console.log "Got warranty info for #{item.model} (#{item.serialNo})...", res.data
-      shipDate = res.data.AssetWarrantyResponse[0].AssetHeaderData.ShipDate
```

Comment:
This is the bit I'm curious about.  Recording the `shipDate` field has been moved into the `lookupShipDate` method, but otherwise these are largely the same.  Are you trying to get rid of the Job altogether, and rely only on the method?  What's the issue that makes it so the Job should _not_ set the `shipDate` field?

Reply:
With the new replacement plan, having an accurate ship/receipt date is going to become really important for determining who is going to get a new computer in the coming year, and going forward, I'm going to ask the students to enter a ship date for all newly received items.  For the most part, the ship date pulled with the Job is accurate, but occasionally one fails to pull (maybe typos in the service tag--I haven't done too much digging) or in the case of some of the monitors, it pulls 12/31/69 as the date.  I don't want the students to have to know that they need to enter a ship date for all Apple products and maybe Dell monitors but not for Dell computers.  Having the lookup method means they can check if there is valid ship date information and if not, then go ahead and supply a date (which shouldn't be overwritten later by the Job). 

I was thinking I would leave the Job to pull warranty information if available because I know Scott uses that feature.

Reply:
Okay, sounds good.  The main purpose of putting things in the job queue is to limit the amount of retrieval happening at any given time, thus respecting any rate limits the remote API server might have in place, as well as keeping the Inventory app responsive.  As long as there's no way to accidentally trigger a large batch of `lookupShipDate` calls at once, this shouldn't be a big deal.